src/core/jbig2_stream.js

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2012 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isDict, isStream } from './primitives';
+import { DecodeStream } from './stream';
+import { Jbig2Image } from './jbig2';
+import { shadow } from '../shared/util';
+
+/**
+ * For JBIG2's we use a library to decode these images and
+ * the stream behaves like all the other DecodeStreams.
+ */
+var Jbig2Stream = (function Jbig2StreamClosure() {
+  function Jbig2Stream(stream, maybeLength, dict, params) {
+    this.stream = stream;
+    this.maybeLength = maybeLength;
+    this.dict = dict;
+    this.params = params;
+
+    DecodeStream.call(this, maybeLength);
+  }
+
+  Jbig2Stream.prototype = Object.create(DecodeStream.prototype);
+
+  Object.defineProperty(Jbig2Stream.prototype, 'bytes', {
+    get() {
+      // If this.maybeLength is null, we'll get the entire stream.
+      return shadow(this, 'bytes', this.stream.getBytes(this.maybeLength));
+    },
+    configurable: true,
+  });
+
+  Jbig2Stream.prototype.ensureBuffer = function() {
```

Comment:
Looks like this was a hack: can we add `req` parameter and probably a comment that we just ignoring interface and moving all parsed data into the buffer.

Reply:
Since this is a pre-existing issue, which is shared between `JpegStream`, `JpxStream`, and `Jbig2Stream`, could we perhaps do that in a follow-up instead?

The interface of all of those streams look kind of weird, and I'm actually a bit surprised that there hasn't been any errors because of it. For example: None of them actually implement `readBlock` methods, and it seems more luck that anything else that we're not calling `getBytes()` (without providing a length) for those streams, since that would trigger a code-path in [`getBytes`]( https://github.com/mozilla/pdf.js/blob/d71a576b302c6eeaa66b2affc055c796b4bd9399/src/core/stream.js#L193-L216) that assumes `readBlock` to exist.

One simple solution might to just replace `ensureBuffer` with `readBlock` for the `JpegStream`, `JpxStream`, and `Jbig2Stream` streams, which seems more correct to me!?
Anyway, since this is an old issue, I'm hoping this point can be deferred to another PR.